pandas/core/internals/blocks.py

```diff
@@ -2978,6 +2978,29 @@ def where(self, other, cond, align=True, errors='raise',
                                                axis=axis, transpose=transpose)
         return result
 
+    def replace(self, to_replace, value, inplace=False, filter=None,
+                regex=False, convert=True):
+        if filter is None or not self.mgr_locs.isin(filter).any():
+            result = self if inplace else self.copy()
+            categories = result.values.categories.tolist()
+            if to_replace in categories:
+                if isna(value):
+                    result.values.remove_categories(to_replace, inplace=True)
+                else:
+                    index = categories.index(to_replace)
+                    categories[index] = value
+                    result.values.rename_categories(categories, inplace=True)
+            if convert:
+                return result.convert(by_item=True, numeric=False,
+                                      copy=not inplace)
+            else:
+                return result
+        else:
+            if not isna(value):
+                self.values.add_categories(value, inplace=True)
```

Comment:
shouldnt this depend on the `inplace` kwarg?

Reply:
fixed